rfc/0054-sdk3-management-apis.md

```diff
@@ -3068,7 +3070,47 @@ Nothing
 
 ### URI
 
-    POST http://localhost:8091/pools/default/buckets/<bucket>/collections/<scope_name> -d name=<collection_name> -d maxTTL=<maxExpiry>
+    POST http://localhost:8091/pools/default/buckets/<bucket>/collections/<scope_name> -d name=<collection_name> -d maxTTL=<maxExpiry> -d history=<history>
+
+## UpdateCollection
+
+Updates an existing collection.
+
+### Signature
+
+```
+void UpdateCollection(CollectionSpec collection, [options])
```

Comment:
We can add overloads in languages that support them, but in languages that don't I think that we're stuck on this path. We can't even just put History in the options because we need to return it as part of the CollectionSpec for Get.

Reply:
can we implement UpdateCollection as ideal API now? we could also provide some overrides and deprecate not-so ideal drop/create

Reply:
> can we implement UpdateCollection as ideal API now? we could also provide some overloads and deprecate not-so ideal drop/create

I'll update the RFC with the new API and mark the old one deprecated. Some SDKs (without overloads) will have to implement the deprecated API for update to be consistent with the other functions.

Reply:
Following team meeting last week I have now updated the RFC with the approach that we settled on.